File: week-02/day-01/draw_diagonal.py
# Write a program that reads a number from the standard input, then draws a
# square like this:
#
#
# %%%%%
# %%  %
# % % %
# %  %%
# %   %
# %%%%%
#
# The square should have as many lines as the number was
a = int(input("Enter a number: "))

if a < 4:
    print("Too small!")
else:
    for i in range(a):
        print("%", end="")
    print()
    for j in range(a - 2):
        print("%", end="")
        for i in range(j):
            print(" ", end="")
        print("%", end="")
        for i in range(a - 2 - j - 1):
            print(" ", end="")
        print("%")
    for i in range(a):
        print("%", end="")

File: week-07/day-1/draw-diagonal.js
'use strict';

var lineCount = 6;


// Write a program that draws a
// square like this:
//
//
// %%%%%
// %%  %
// % % %
// %  %%
// %   %
// %%%%%
//
// The square should have as many lines as lineCount is
for(let i = 0; i < lineCount; i++){
    let line = '';
    if(i === 0 || i === lineCount - 1){
        for(let j = 0; j < lineCount; j++){
            line += '%';
        }
    } else {
        line += '%';
        for(let j = 0; j < lineCount - 2; j++){
            if(j === i - 1){
                line += '%';
            } else {
                line += ' ';
            }
        }
        line += '%';
    }
    console.log(line);
}